src/test/rustdoc/macro_pub_in_module.rs

```diff
@@ -1,49 +1,73 @@
 // aux-build:macro_pub_in_module.rs
 // edition:2018
 // build-aux-docs
-// @has external_crate/some_module/macro.external_macro.html
 
 //! See issue #74355
 #![feature(decl_macro, no_core, rustc_attrs)]
 #![crate_name = "krate"]
 #![no_core]
 
+// @has external_crate/some_module/macro.external_macro.html
 extern crate external_crate;
 
 pub mod inner {
     // @has krate/inner/macro.raw_const.html
+    // @!has krate/macro.raw_const.html
     pub macro raw_const() {}
 
     // @has krate/inner/macro.test.html
+    // @!has krate/macro.test.html
     #[rustc_builtin_macro]
     pub macro test($item:item) {}
 
     // @has krate/inner/macro.Clone.html
+    // @!has krate/macro.Clone.html
     #[rustc_builtin_macro]
     pub macro Clone($item:item) {}
 
-    // Make sure the logic is not affected by a re-export.
+    // Make sure the logic is not affected by re-exports.
+    mod unrenamed {
+        // @!has krate/macro.unrenamed.html
+        #[rustc_macro_transparency = "semitransparent"]
+        pub macro unrenamed() {}
+    }
+    // @has krate/inner/macro.unrenamed.html
+    pub use unrenamed::unrenamed;
```

Comment:
This check fails (and in practice, matches the case of `::core::task::ready!` re-exported from a private `mode ::core::task::ready`) because of:

https://github.com/rust-lang/rust/blob/76188b6830897a875a1289809c12b5c5d69bb7ef/compiler/rustc_privacy/src/lib.rs#L834-L840

Indeed, `rustdoc` uses the public-reachability info to decide whether or not to inline the definition of the item when it is left unspecified.

https://github.com/rust-lang/rust/blob/76188b6830897a875a1289809c12b5c5d69bb7ef/src/librustdoc/visit_ast.rs#L179-L184

If `rustc` says the item is always publicly reachable, then `rustdoc` decides not to inline the definition, which is thus then missing when the private items are stripped.

I have hacked my way around this limitation with https://github.com/rust-lang/rust/pull/77862/commits/73369151b4278ca665494d449c3e4d8e1278d5ba, but I'd welcome some feedback regarding what is deemed the best course of action, here.

  - cc @jyn514 @petrochenkov